src/metabase/driver/postgres/actions.clj

```diff
@@ -27,38 +23,75 @@
           (map :column_name)
           (jdbc/reducible-query jdbc-spec sql-args {:identifers identity, :transaction? false}))))
 
-(defn- maybe-parse-unique-constraint-error [database error-message]
-  (let [[match? constraint _value]
-        (re-find #"ERROR:\s+duplicate key value violates unique constraint \"([^\"]+)\"" error-message)]
-    (when match?
-      (let [columns (constraint->column-names database constraint)]
-        (mapv
-         (fn [column]
-           {:message (tru "violates unique constraint {0}" constraint)
-            :constraint constraint
-            :column column})
-         columns)))))
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:postgres actions.error/violate-not-null-constraint]
+  [_driver error-type _database _action-type error-message]
+  (when-let [[_ column]
+             (re-find #"null value in column \"([^\"]+)\".*violates not-null constraint"  error-message)]
+    {:type    error-type
+     :message (tru "{0} must have values." (str/capitalize column))
+     :errors  {column (tru "You must provide a value.")}}))
+
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:postgres actions.error/violate-unique-constraint]
+  [_driver error-type database _action-type error-message]
+  (when-let [[_match constraint _value]
+             (re-find #"duplicate key value violates unique constraint \"([^\"]+)\"" error-message)]
+    (let [columns (constraint->column-names database constraint)]
+      {:type    error-type
+       :message (tru "{0} already exist." (u/build-sentence (map str/capitalize columns) :stop? false))
```

Comment:
This needs to be "exists" when there is one column. You can use `trun` instead of `tru` for that.

Reply:
that's a bit tricky to use. [ecfde21](https://github.com/metabase/metabase/pull/32935/commits/ecfde213405235d7af83110b062b67654b1e87cf)